src/phoenix/server/api/routers/v1/traces.py

```diff
@@ -31,7 +33,23 @@
 from .pydantic_compat import V1RoutesBaseModel
 from .utils import RequestBody, ResponseBody, add_errors_to_responses
 
-router = APIRouter(tags=["traces"])
+
+async def validate_api_key(request: Request) -> None:
+    """
+    Validates the API key in the request headers.
+
+    Args:
+        request: The incoming request.
+    Returns:
+        None
+    """
+    if (validate := request.app.state.validate_api_key) and (
+        not ((api_key := request.headers.get(AUTH_HEADER)) and validate(api_key))
+    ):
+        raise HTTPException(status_code=HTTP_403_FORBIDDEN, detail="Invalid API key")
+
+
+router = APIRouter(tags=["traces"], dependencies=[Depends(validate_api_key)])
```

Comment:
for now i need the graphql to be unlocked in order to get the api key, otherwise i would need to somehow put an api key into the database, and that's not so easy